### PR TITLE
Fix media attachment order of remote posts

### PR DESCRIPTION
**Pull request by ClearlyClaire**
Fixes a regression introduced by #28043

cc @mjankowski

### CODE REVIEW

**Review by timetinytim — DISMISSED** [review]

**Review by renchap — DISMISSED** [review]

File: spec/lib/activitypub/activity/create_spec.rb [hunk 1]
```diff
@@ -532,15 +532,21 @@ def activity_for_object(json)
...
           }
         end
 
-        it 'creates status' do
+        it 'creates status with correctly-ordered media attachments' do
           status = sender.statuses.first
 
           expect(status).to_not be_nil
-          expect(status.media_attachments.map(&:remote_url)).to include('http://example.com/attachment.png')
+          expect(status.ordered_media_attachments.map(&:remote_url)).to eq ['http://example.com/attachment.png', 'http://example.com/emoji.png']
```
Comment (mjankowski) [inline spec/lib/activitypub/activity/create_spec.rb:548]: I take it that the order we want to preserve here is "the order that the document types showed up in the json"?

Looks fine overall.

Comment (ClearlyClaire) [reply spec/lib/activitypub/activity/create_spec.rb:548]: yes

**Comment by ClearlyClaire** [comment]
I really don't understand the CI failures :|

**Comment by ClearlyClaire** [comment]
Ah, I think that's some issue with the status not being persisted when the check is done, and thus the collection proxy working, but anything more complex would hit the non-updated-database yet and fail. This is a very annoying quirk, I'll see what I can do.

**Review by renchap — APPROVED** [review]

**Pull request merged**